### DISCUSSION TITLE
Error in running a test after adding a consumer

### DISCUSSION THREAD

**Discussion by sahil-kaushik1** [Cyfrin/foundry-full-course-cu#2818] [Q&A] [answered]
function testplayerwhennotopen() public {
        vm.prank(PLAYER);
        raffle.EnterRaffle{value: raffleEntranceFee}();
        raffle.performUpkeep("");
        vm.warp(block.timestamp + automationUpdateInterval + 1);
        vm.roll(block.number + 1);
        vm.expectRevert(Raffle.Raffle_Notopen.selector);
        vm.prank(PLAYER);
        raffle.EnterRaffle{value: raffleEntranceFee}();
    }
    
  function performUpkeep(bytes calldata /* performData */) external {
        (bool unkeepneeded, ) = KeepCheckUp("");
        if (!unkeepneeded) {
            revert Raffle_Notopen();
        }
        if ((block.timestamp - s_lasttimestamp) < i_interval) {
            revert();
        }
        raffleflag = RaffleState.CALCULATING;
        uint256 requestId = s_vrfCoordinator.requestRandomWords(
            VRFV2PlusClient.RandomWordsRequest({
                keyHash: i_keyHash,
                subId: i_subId,
                requestConfirmations: REQUESTCONFIRMATIONS,
                callbackGasLimit: i_callbackGasLimit,
                numWords: NUMWORDS,
                extraArgs: VRFV2PlusClient._argsToBytes(
                    // Set nativePayment to true to pay for VRF requests with Sepolia ETH instead of LINK
                    VRFV2PlusClient.ExtraArgsV1({nativePayment: false})
                )
            })
        );
    }
    
contract AddConsumer is Script {
    function run() external {
        address mostrecentdeployed=DevOpsTools.get_most_recent_deployment("Raffle",block.chainid);
        Addconsumerusingconfig(mostrecentdeployed);

    }
    function Addconsumerusingconfig( address adresstovrf) public {
        HelperConfig helperconfig=new HelperConfig();
        address vrfcoordinator = helperconfig.getConfig().s_vrfCoordinator;
        uint256 subId = helperconfig.getConfig().subId;
        Addconsumer(adresstovrf, vrfcoordinator, subId);
        
    }
    function Addconsumer(address adresstovrf,address vrf , uint256 subid) public {
        console.log("Adding consumer contract: ",adresstovrf);
        console.log("Using VRFCoordinator: ", vrf);
        console.log("On chain id: ", block.chainid);

        vm.startBroadcast();
        VRFCoordinatorV2_5Mock(vrf).addConsumer(subid,adresstovrf);
        vm.stopBroadcast();
    }

}

i found  all the code correct and infact it is giving me a revert but still the test is failing . if any one knows the solution comment it with a n explanation.

thanks for ur help
![Screenshot from 2024-11-06 01-39-08](https://github.com/user-attachments/assets/c02c88e3-7008-46c8-ba4a-3d06eee2b270)

**Comment by EngrPips** [comment]
Hello @sahil-kaushik1, are you writing this test along with Patrick?

**Reply by sahil-kaushik1** [reply]
Yes I was following him

**Reply by EngrPips** [reply]
Please share the link to the lesson you are currently watching.

**Reply by sahil-kaushik1** [reply]
Here is the link 
https://updraft.cyfrin.io/courses/foundry/smart-contract-lottery/add-consumer

**Reply by EngrPips** [reply]
I am not sure why the video is not loading for me, but here https://github.com/Cyfrin/foundry-smart-contract-lottery-cu/blob/main/test/unit/RaffleTest.t.sol is the link to the course test file. Compare the code against yours to see what you are doing wrongly.

**Reply by sahil-kaushik1** [reply]
I will notify once the error is found

**Reply by EngrPips** [reply]
Alright, I am looking forward to hearing from you.

**Reply by sahil-kaushik1** [reply, marked as answer]
yo the error resolved .
Actually the there was problem in the sequence of code executing .
i debugged it , it was so fun btw 
finally i can relax and move to next topic.

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.
